tiatoolbox/models/engine/engine_abc.py

```diff
@@ -571,9 +578,14 @@ def infer_patches(
                 batch_data["image"],
```

Comment:
This is where I have a problem with the design! (I guess I mentioned this when we doing ModelABC).

I think we should get rid of ModelABC altogether, if possible! Instead of `self.model.infer_batch`, we should call `self.infer_batch`, in other words, `infer_batch` should be a method of the engine, not the model! The same goes with the `preproc_func` and `postproc_func`. These are static methods anyway. 

The reason I am suggesting this is to make the engines more versatile and the whole toolbox easier to use with new models. I understand that we may end up needing new modules for loading preprocessing, postprocessing, and weight loading of the pretrained models (model zoo), but that also maybe make it more organized and help avoid repetition. If we are going to do this, it is better to be done early rather than late.

Reply:
We will need a separate PR to update this as discussed.